python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -250,27 +250,26 @@ def launch_scheduled_runs(
                 )
             error_locations.add(location_entry.origin.location_name)
 
-    # Remove any schedule states that were previously created with DECLARED_IN_CODE
-    # and can no longer be found in the workspace (so that if they are later added
-    # back again, their timestamps will start at the correct place)
+    # Remove any schedule states that can no longer be found in the workspace
+    # (if they are later added back again, their timestamps will start at the correct place)
     states_to_delete = [
         schedule_state
         for selector_id, schedule_state in all_schedule_states.items()
         if selector_id not in schedules
-        and schedule_state.status == InstigatorStatus.DECLARED_IN_CODE
     ]
     for state in states_to_delete:
         location_name = state.origin.external_repository_origin.code_location_origin.location_name
-        # don't clean up auto running state if its location is an error state
+        # don't clean up state if its location is an error state
         if location_name not in error_locations:
```

Comment:
We test that schedules are not deleted from locations with load errors in `test_status_in_code_schedule` https://github.com/dagster-io/dagster/blob/650fc8f231b6c1ab7323360e50f975af813d836d/python_modules/dagster/dagster_tests/scheduler_tests/test_scheduler_run.py#L1013-L1026